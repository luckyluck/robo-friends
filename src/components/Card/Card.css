.Card {
    width: calc(25% - 16px);
}

@media (max-width: 1024px) {
    .Card {
        width: calc(100% / 3 - 16px);
    }
}

@media (max-width: 767px) {
    .Card {
        width: calc(100% / 2 - 16px);
    }
}

@media (max-width: 500px) {
    .Card {
        width: calc(80% - 16px);
    }
}
