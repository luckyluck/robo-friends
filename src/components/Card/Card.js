import React from 'react';

import './Card.css';

const Card = ({ id, name, email }) => {
    return (
        <div className="Card tc bg-light-green dib br3 ma2 grow bw2 shadow-5 pv1 ph3">
            <img src={`https://robohash.org/${id}?size=200x200`} alt={name}/>
            <div>
                <h2>{name}</h2>
                <p>{email}</p>
            </div>
        </div>
    );
};

export default Card;