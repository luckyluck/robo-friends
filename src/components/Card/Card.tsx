import * as React from 'react';

import './Card.css';

interface ICardProps {
    id: string,
    name: string,
    email: string
}

const Card: React.SFC<ICardProps> = ({ id, name, email }) => {
    return (
        <div className="Card tc bg-light-green dib br3 ma2 grow bw2 shadow-5 pv1 ph3">
            <img src={`https://robohash.org/${id}?size=200x200`} alt={name}/>
            <div>
                <h2>{name}</h2>
                <p>{email}</p>
            </div>
        </div>
    );
};

export default Card;