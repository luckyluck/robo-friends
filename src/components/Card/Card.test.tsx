import { shallow } from 'enzyme';
import * as React from 'react';
import Card from './Card';

describe('Card component', () => {
    const defaultProps = {
        email: '',
        id: '',
        name: ''
    };
    it('should shallowly render component', () => {
        expect(shallow(<Card {...defaultProps}/>)).toMatchSnapshot();
    });

    it('should shallowly render 1 component', () => {
        expect(shallow(<Card {...defaultProps}/>).length).toEqual(1);
    });
});