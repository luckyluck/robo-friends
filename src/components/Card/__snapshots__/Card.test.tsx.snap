// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should shallowly render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    email=""
    id=""
    name=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="https://robohash.org/?size=200x200"
        />,
        <div>
          <h2>
            
          </h2>
          <p>
            
          </p>
        </div>,
      ],
      "className": "Card tc bg-light-green dib br3 ma2 grow bw2 shadow-5 pv1 ph3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "https://robohash.org/?size=200x200",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              
            </h2>,
            <p>
              
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="https://robohash.org/?size=200x200"
          />,
          <div>
            <h2>
              
            </h2>
            <p>
              
            </p>
          </div>,
        ],
        "className": "Card tc bg-light-green dib br3 ma2 grow bw2 shadow-5 pv1 ph3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "https://robohash.org/?size=200x200",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                
              </h2>,
              <p>
                
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
